### PR TITLE
looking at the insert route

### PR DESCRIPTION
**Pull request by manandearth**
The aim is to insert entries to the users table.
I see the route `"/invoices" :get ,,, query-params`. 
I only manage to use this route if I disable the `"/invoices/:id"` route.
Do I need to make a new route '"/invoices/:id/insert"` that will accept query-params, or a form with a :post method?
I like the spec by the way!

### CODE REVIEW

File: src/playground/service.clj [hunk 1]
```diff
@@ -74,9 +74,9 @@
     ["/about" :get (conj common-interceptors `about-page)]
     ["/api" :get (into component-interceptors [http/json-body (param-spec-interceptor ::api :query-params) `api])]
     ["/invoices/insert" :get (into component-interceptors [http/json-body (param-spec-interceptor ::invoices.insert/api :query-params) `invoices.insert/perform])]
-    ["/invoices/:id" :get (into component-interceptors [http/json-body (param-spec-interceptor ::invoices.retrieve/api :path-params) `invoices.retrieve/perform])]
+    #_["/invoices/:id" :get (into component-interceptors [http/json-body (param-spec-interceptor ::invoices.retrieve/api :path-params) `invoices.retrieve/perform])]
```
Comment (manandearth) [inline src/playground/service.clj]: ignoring this form allows : `http://localhost:8080/invoices/insert?amount=22` to be inserted to the table.

Comment (vemv) [reply src/playground/service.clj]: But such routing isn't twice here https://github.com/reducecombine/playground/blob/master/src/playground/service.clj

Routes can have exactly one entry per path (`/invoices/:id`) and method (`POST`), else how would the router know what option to pick?

Comment (vemv) [reply src/playground/service.clj]: Ah no sorry it's not twice, misread the diff. Ignore my comment

**Comment by vemv** [comment]
Can reproduce. Will fix upstream (i.e. in the original repo) and let you know.

**Comment by vemv** [comment]
Hey there,

this was fun and useful to research. I'm no Pedes expert so I'm happy to improve my own setup.

First of all there's the issue that in `playground`, routes don't ever get reloaded in the system, right?

I do _not_ have that problem in my more advanced template, which is closed-source atm.

Then, even assuming you can reload the routes, the problem boils down to having two similar routes which create an ambiguity.

This is described in http://pedestal.io/guides/defining-routes#_constraints and particularly http://pedestal.io/guides/defining-routes#_considering_constraints .

The fix would be to change the routes definition format from `(def routes #{...})` to `(def routes (io.pedestal.http.route.definition.table/table-routes ...))`.

But such migration involves a few steps. I don't want to update `playground` with new boilerplate. The private project, once open-sourced, will supersede `playground` next week, and will support both kind of formats.

So, long story short:

* work around the conflict by changing `"/invoices/insert"` to `"/invoices-insert"`
* Add a clear FIXME comment so you remember to change it later
* Later you'll leave this fork behind and redo your work.
  * Will be quick, I promise
  * Keep commits short and clean as indicated, so you redo work effectively.

**Comment by manandearth** [comment]
Will do . Thanks. 
Might have a bit of cleaning-up kind of mess on the next commit.. If you see what I mean.
